src/__tests__/utils.js

```diff
@@ -1,216 +1,246 @@
+    expect(matchFileType(undefined, 'test.js')).toBe(false);
+    expect(matchFileType(null, 'test.js')).toBe(false);
+    expect(matchFileType({}, 'test.js')).toBe(false);
+  });
 
-  // If an ESLint rule is disabled fall back to prettier defaults.
-  { rules: { "max-len": [0, { code: 120 }] }, options: {} },
-  { rules: { quotes: ["off", { code: 120 }] }, options: {} },
-  { rules: { quotes: ["backtick", { code: 120 }] }, options: {} },
-  { rules: { semi: "off" }, options: {} },
-  { rules: { semi: ["off", "never"] }, options: {} },
-  { rules: { semi: ["warn", "always"] }, options: {} },
-  { rules: { semi: ["warn", "always"] }, options: { semi: true } },
-  { rules: { semi: ["error", "never"] }, options: { semi: false } },
-  { rules: { semi: [2, "never"] }, options: { semi: false } },
-  { rules: { semi: [2, "never"] }, options: { semi: false } },
-  { rules: { indent: "off" }, options: {} },
-  { rules: { indent: ["off", "tab"] }, options: {} },
-  { rules: { indent: ["warn", 2] }, options: { useTabs: false, tabWidth: 2 } },
-  { rules: { indent: ["warn", 4] }, options: { useTabs: false, tabWidth: 4 } },
-  { rules: { indent: ["error", "tab"] }, options: { useTabs: true } },
-  { rules: { indent: [2, "tab"] }, options: { useTabs: true } }
-];
+  test("returns false if fileName is not a string", () => {
+    expect(matchFileType('JavaScript', undefined)).toBe(false);
+    expect(matchFileType('JavaScript', null)).toBe(false);
+    expect(matchFileType('JavaScript', 123)).toBe(false);
+    expect(matchFileType('JavaScript', {})).toBe(false);
+  });
 
-getPrettierOptionsFromESLintRulesTests.forEach(
-  ({ rules, options, prettierOptions, fallbackPrettierOptions }, index) => {
-    test(`getPrettierOptionsFromESLintRulesTests ${index}`, () => {
-      const { prettier } = getOptionsForFormatting(
-        { rules },
-        prettierOptions,
-        fallbackPrettierOptions
-      );
-      expect(prettier).toMatchObject(options);
-    });
-  }
-);
+  test("returns false if fileName doesn't match", () => {
+    expect(matchFileType('JavaScript', 'test.css')).toBe(false);
+    expect(matchFileType('CSS', 'test.js')).toBe(false);
+  });
 
-test("if prettierOptions are provided, those are preferred", () => {
-  const { prettier } = getOptionsForFormatting(
-    { rules: { quotes: [2, "single"] } },
-    {
-      singleQuote: false
-    }
-  );
-  expect(prettier).toMatchObject({ singleQuote: false });
+  test("returns true if fileName matches", () => {
+      expect(matchFileType('JavaScript', 'test.js')).toBe(true);
+      expect(matchFileType('JavaScript', 'test.es')).toBe(true);
+      expect(matchFileType('CSS', 'test.css')).toBe(true);
+      expect(matchFileType('Less', 'test.less')).toBe(true);
+  });
 });
 
-// eslint-disable-next-line max-len
-test(`if fallbacks are provided, those are preferred over disabled eslint rules`, () => {
-  const { prettier } = getOptionsForFormatting(
+describe("getOptionsForFormatting", () => {
```

Comment:
Sorry for a mess in this file, just wrapped tests with a `describe` block.

Reply:
I think it sounds good to do this in a separate PR, we can work in a refactoring PR together once we've solved this problem.